### ISSUE TITLE
Default Values for models

### ISSUE THREAD

**Issue by dupski** [RevJS/revjs#16]
How do we do default field values?

maybe add a 'defaults()' function to IModel?

Models can then implement

defaults() {
  this.status = draft;
  this.start_date = .... etc
}

The alternative is to have a "default" option for each field decorator, which could either take a value or function. I like the above option more though :)

**Comment by dupski** [comment]
This should **definitely not** be with values assigned to fields in the constructor because these get automatically assigned when a model is rehydrated on the Server, thereby adding data that was not sent by the client.

We should probably add an explicit check (maybe when models are registered?) that no instance values are set when a model is hydrated (maybe using Object.values() - we'll need to consider property getters when doing this check)

Suggest the defaults process works as follows:

Create a 'defaults()' function for model classes (which could take IMethodContext for useful contextual information), e.g.

```ts
    defaults(ctx: IMethodContext) {
        if ('default_type' in ctx.args) {
            this.type = 'default_type';
        }
        else {
            this.type = 'organisation';
        }
    }
```
This method can then be used by calling `manager.getNew(ModelClass)`, which returns a new instance of the model, with the default values applied. In the future, this method can take context info which can also be passed down to the detauls() method.

Work Items:

* [x] - Add check when registering models that they don't set field values in the constructor
* [x] - Test what happens with property getters with the above change
* [x] - Add optional `defaults()` to `IModel` interface
* [x] - Add `getNew()` method for `ModelManager`

**Comment by dupski** [comment]
Released in v0.19.0
